### PR TITLE
Fixes #219: Notifier help page improvements

### LINKED ISSUE #219: Notifier help page improvements
Would you please update

https://docs.zen-cart.com/dev/code/notifiers/

and give that article a scrub and improve the examples?  Thank you!

### PR DESCRIPTION
**Pull request by lat9**

### CODE REVIEW

File: content/dev/code/notifiers.md [hunk 1]
```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
...
+One of the many goals of the Zen Cart project has always been to make it simple for third party developers to add functionality to the core code in an easy and unobtrusive manner. To do this we use the [override system](/user/template/template_overrides/), the auto inclusion system and the observer-notifier system.
 
-The observer/notifier system is an implementation of the ["pub-sub" pattern](https://en.wikipedia.org/wiki/Publish–subscribe_pattern).  It was introduced to give developers deep access to core operation, without the need to touch any core files at all. Although it was written for an object-oriented code base, it can be used with procedural code as well.
+The observer/notifier system is an implementation of the ["pub-sub" pattern](https://en.wikipedia.org/wiki/Publish–subscribe_pattern)  that was introduced to give developers deep access to core operation without the need to touch any core files at all. Although the implementation was written for an object-oriented code base, it can alse be used with procedural code.
```
Comment (drbyte) [inline content/dev/code/notifiers.md:13]: > it can alse be used with

Typo?

Comment (scottcwilson) [reply content/dev/code/notifiers.md:13]: Yep.  Will get it on the next checkin.

File: content/dev/code/notifiers.md [hunk 2]
```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
...
+    c. [Event-Specific Update Methods](event-specific-update-methods).
 
-Events are triggered by calling `$this->notify` in any class that `extends base`:
+5. [Additional Information](additional-information).  This section has references to additional documentation on the observer/notifier system.
 
-<pre> $this->notify('EVENT_NAME');
-</pre>
 
-Example: In the shopping cart class after an item has been added to the cart this event is triggered:
+## Issuing Event Notifications
```
Comment (drbyte) [inline content/dev/code/notifiers.md:34]: I feel like the word "issue" doesn't fully capture what's happening. Perhaps "trigger" would be more clear, and allow easier wording later about "triggering and issuer" which could be replaced with "triggered".

File: content/dev/code/notifiers.md [hunk 3]
```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
...
 
-<pre> $this->notify('NOTIFIER_CART_ADD_CART_END');
-</pre>
+The point of the observer/notifier system is to enable developers to write code that listens for certain events to happen and then perform a customized action for an event.
 
-In procedural code (or outside of a class that `extends base`) use the global `$zco_notifier` object. For example, inside the `zen_mail` function this event is triggered, which allows a plugin to update the sent email format:
+Events are identified by a string-names and are triggered via call to a `notify` method by an _event issuer_.  You can see a [list of notifiers](/dev/code/notifiers_list) provided by the Zen Cart core for reference.
```
Comment (drbyte) [inline content/dev/code/notifiers.md:38]: > identified by a string-names

Perhaps remove "a" ?

File: content/dev/code/notifiers.md [hunk 4]
```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
...
 
-### Observe and Prosper
+The `notify` method takes the following inputs:
 
-To take advantage of notifiers, developers need to write some code to watch for them. Observers need to be written as a class. There's even a nice directory, `includes/classes/observers`, where developers can put these classes.
+|    Input Name     | Required? | Description                                                  |
+| :---------------: | :-------: | ------------------------------------------------------------ |
+|     $eventId      |    Yes    | The string 'name' of the event, e.g. `NOTIFIER_CART_ADD_CART_END`. |
+|      $param1      |    No     | A read-only variable, the format of which varies by the `$eventId`.  Defaults to an empty array. |
+| $param2 - $param9 |    No     | A collection of read-write variables, passed as a reference.  The notification issuer is, essentially, giving permission for an observer to update these variable.  Each variable's format (and presence) varies by the `$eventId` and defaults to `null`. |
```
Comment (drbyte) [inline content/dev/code/notifiers.md:48]: > permission for an observer to update these variable.  

`variable` -> `variables` ?

File: content/dev/code/notifiers.md [hunk 5]
```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
...
 ```
 
-As you can see we have defined a new class called myObserver and in the constructor function for that class (function myObserver) have attached this myObserver class to the event `NOTIFIER_CART_ADD_CART_END`.
+For example, the shopping cart class issues this event after an item has been added to the cart:
+```php
+$this->notify('NOTIFIER_CART_ADD_CART_END');
+```
 
-"Fine," I hear you saying, "but how do I actually do anything useful?"
+When a class issues a notification using the `$this` keyword, _all_ **public** class-variables are available for use by a watching observer.  If a class desires to issue a notification without that access, the global `$zco_notifier` can be used to issue that notification.
```
Comment (drbyte) [inline content/dev/code/notifiers.md:69]: I'm not sure I understand the 2nd sentence here.

Comment (lat9) [reply content/dev/code/notifiers.md:69]: I agree, now reviewing what was written.  An observer could always use a `global` on the issuing class to retrieve those public variables.

File: content/dev/code/notifiers.md (hunk 4 above)
Comment (lat9) [reply content/dev/code/notifiers.md:48]: Agreed.

File: content/dev/code/notifiers.md (hunk 2 above)
Comment (lat9) [reply content/dev/code/notifiers.md:34]: Agreed.  Just remember that there are intra-page links at the top of the document, so those would need changing as well if the level-2-heading text is changed.

File: content/dev/code/notifiers.md (hunk 5 above)
Comment (drbyte) [reply content/dev/code/notifiers.md:69]: Agreed. Might also need to `global` on both ends if the calling source doesn't already have access to global scope.

**Comment by drbyte** [comment]
@lat9 Thanks for taking the time to improve these docs! Much appreciated!

**Pull request merged**